orm/versioning_bucket.go

```diff
@@ -53,7 +53,7 @@ func (b VersioningBucket) GetLatestVersion(db weave.ReadOnlyKVStore, id []byte)
 	for _, m := range matches {
 		var vID VersionedIDRef
 		idData := m.Key[dbKeyLength:]
-		if err := vID.Unmarshal(idData); err != nil {
+		if err = vID.Unmarshal(idData); err != nil {
```

Comment:
The previous version was correct, as this is supposed to be a scoped `err` variable.

Everywhere in this file, you are changing from the creation of a local `err` variable to reusing the same `err` declaration. I think this is a mistake as the original construction is correct.

Reply:
yeah, I don't like this one as much either, need to see if we can configure ignoring shadowing, but enforcing checking

Reply:
If you call it eg `topErr` in the top level, and `err` in the if statements, you can have the separation and keep the linter happy.

I agree that can be a bit annoying, but seeing what else it caught, maybe we just make a habit. Like

`if irr :== foo(); irr != nill { ... }`

`irr` for `if-err`